src/pip/_internal/index/package_finder.py

```diff
@@ -585,14 +571,13 @@ class PackageFinder:
 
     def __init__(
         self,
-        link_collector,       # type: LinkCollector
-        target_python,        # type: TargetPython
-        allow_yanked,         # type: bool
-        format_control=None,  # type: Optional[FormatControl]
-        candidate_prefs=None,         # type: CandidatePreferences
-        ignore_requires_python=None,  # type: Optional[bool]
-    ):
-        # type: (...) -> None
+        link_collector: LinkCollector,
+        target_python: TargetPython,
+        allow_yanked: bool,
+        format_control: Optional[FormatControl] = None,
+        candidate_prefs: Optional[CandidatePreferences] = None,
```

Comment:
I annotated `candidate_prefs` as `Optional[CandidatePreferences] = None`, when the original annotation was `CandidatePreferences`.

This happened before. Mypy is not on "strict" mode, so it doesn't fails here.

Reply:
We want to change this on #10091. But I think it must take more time before we fix this.